src/core/public/application/ui/app_router.tsx

```diff
@@ -83,18 +79,15 @@ export const AppRouter: FunctionComponent<Props> = ({
               url,
             },
           }: RouteComponentProps<Params>) => {
-            // Find the mounter including legacy mounters with subapps:
-            const [id, mounter] = mounters.has(appId)
-              ? [appId, mounters.get(appId)]
-              : [...mounters].filter(([key]) => key.split(':')[0] === appId)[0] ?? [];
-
+            const [id, mounter] = mounters.has(appId) ? [appId, mounters.get(appId)] : [];
```

Comment:
I initially totally removed the id retrieval logic from the catch-all handler, and was just passing an undefined mounter and a 'not-found' id to `AppContainer`, however this catch-all is still used to properly dispaly the `uptime` app, due to the way they declare their base path:

https://github.com/elastic/kibana/blob/532f2d70e84af2aec4df06331643fbb6e0ba5033/x-pack/plugins/uptime/public/apps/plugin.ts#L86-L89

Not sure how or why, but the hashbang in the appRoute make their specific route not matching, the previous routes definition, but the appId is properly identified by the catch-all.

keeping these 3 lines do not have a significant impact anyway, so I did not spend much time trying to fix an issue around hashbang routing, that we officially are not supporting anyway.

Reply:
Can we open an issue for the Uptime team to migrate away from hashbang routing and leave a note that we can remove these lines once that is done?

Reply:
Created https://github.com/elastic/kibana/issues/76348 and will add comment.